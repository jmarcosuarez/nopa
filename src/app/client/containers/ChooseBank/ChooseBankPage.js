/* eslint-disable global-require */
import React from 'react';
import { Layout, Button } from '../../components';
import * as Paths from '../../constants/paths';
import styles from './ChooseBank.css';

const bankList = [{
  name: 'Barclays',
  logo: require('../../../static/images/Barclays.png'),
},
{
  name: 'Natwest',
  logo: require('../../../static/images/LogoNatwest.png'),
},
{
  name: 'Lloyds',
  logo: require('../../../static/images/LogoLloyds.png'),
},
{
  name: 'HSBC',
  logo: require('../../../static/images/LogoHSBC.png'),
},
{
  name: 'TSB',
  logo: require('../../../static/images/LogoTSB.png'),
},
{
  name: 'Santander',
  logo: require('../../../static/images/LogoSantander.png'),
},
];

const ChooseBankPage = () => (
  <Layout title="Choose your bank!">
    <div className={styles.main}>
      <h1>Which bank does this account belong to?</h1>
      <p>Track all of your payments by connecting as many bank accounts as you&#39;d like to your Nopa<br />
          account and get updates on your balance instantly.</p>

      <div className={styles.bankList}>
        {
              bankList.map(bank => <div key={bank.name}><img alt={bank.name} src={bank.logo} /></div>)
            }
      </div>

      <Button to={Paths.LOGIN_BANK} className="button">Get started</Button>

    </div>
  </Layout>
  );

export default ChooseBankPage;
