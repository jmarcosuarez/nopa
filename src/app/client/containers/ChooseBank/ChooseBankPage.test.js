/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import ChooseBankPage from './ChooseBankPage';

describe('Renders correctly', () => {
  it('should render correctly', () => {
    const output = shallow(
      <ChooseBankPage />,
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
