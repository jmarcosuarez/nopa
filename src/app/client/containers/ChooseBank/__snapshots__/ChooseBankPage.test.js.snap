// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly should render correctly 1`] = `
<Layout
  title="Choose your bank!"
>
  <div
    className="main"
  >
    <h1>
      Which bank does this account belong to?
    </h1>
    <p>
      Track all of your payments by connecting as many bank accounts as you'd like to your Nopa
      <br />
      account and get updates on your balance instantly.
    </p>
    <div
      className="bankList"
    >
      <BankListItem
        alt="Barclays"
        onSelectBank={[Function]}
        src="test-file-stub"
      />
      <BankListItem
        alt="Natwest"
        onSelectBank={[Function]}
        src="test-file-stub"
      />
      <BankListItem
        alt="Lloyds"
        onSelectBank={[Function]}
        src="test-file-stub"
      />
      <BankListItem
        alt="HSBC"
        onSelectBank={[Function]}
        src="test-file-stub"
      />
      <BankListItem
        alt="TSB"
        onSelectBank={[Function]}
        src="test-file-stub"
      />
      <BankListItem
        alt="Santander"
        onSelectBank={[Function]}
        src="test-file-stub"
      />
    </div>
    <Button
      to="/login"
    >
      Get started
    </Button>
  </div>
</Layout>
`;
